pennylane/capture/primitives.py

```diff
@@ -82,9 +82,94 @@ def _pow(a, b):
     return AbstractOperator
 
 
-def create_operator_primitive(operator_type: type) -> Optional["jax.core.Primitive"]:
+@lru_cache
+def _get_abstract_measurement():
+    if not has_jax:  # pragma: no cover
+        raise ImportError("Jax is required for plxpr.")  # pragma: no cover
+
+    class AbstractMeasurement(jax.core.AbstractValue):
+        """An abstract measurement.
+
+        Args:
+            abstract_eval (Callable): See :meth:`~.MeasurementProcess._abstract_eval`.  A function of
+               ``n_wires``, ``has_eigvals``, ``num_device_wires`` and ``shots`` that returns a shape
+               and numeric type.
+            n_wires=None (Optional[int]): the number of wires
+            has_eigvals=False (bool): Whether or not the measurement contains eigenvalues in a wires+eigvals
+               diagonal representation.
+
+        """
+
+        def __init__(
+            self, abstract_eval: Callable, n_wires: Optional[int] = None, has_eigvals: bool = False
+        ):
+            self._abstract_eval = abstract_eval
+            self._n_wires = n_wires
+            self.has_eigvals: bool = has_eigvals
+
+        def abstract_eval(self, num_device_wires: int, shots: int) -> Tuple[Tuple, type]:
```

Comment:
Is there a distinction between `num_device_wires` and `n_wires` (specifically regarding the "device" part of the name)?

Reply:
Got it, without looking at the qnode pr it wasn't clear to me what `num_device_wires` was about. Makes sense now :)